include/boost/geometry/algorithms/detail/envelope/interface.hpp

```diff
@@ -94,7 +127,18 @@ struct envelope<boost::variant<BOOST_VARIANT_ENUM_PARAMS(T)> >
 template<typename Geometry, typename Box>
 inline void envelope(Geometry const& geometry, Box& mbr)
 {
-    resolve_variant::envelope<Geometry>::apply(geometry, mbr);
+    // TODO put this into a resolve_strategy stage
+    //      (and take the return type from resolve_variant)
```

Comment:
Yes, this should be done (TODO). So in 2 parameter envelope pass object of type `bg::default_strategy` into the `resolve_variant::envelope<>` and specialize for this type above.